app/javascript/elm/src/Main.elm

```diff
@@ -226,6 +231,10 @@ type Msg
     | HighlightSessionMarker (Maybe Location)
     | GraphRangeSelected (List Float)
     | UpdateIsShowingTimeRangeFilter Bool
+    | Timeout Int
+    | UpdateResolution
+    | MaybeIncreaseResolution
+    | MaybeDecreaseResolution
```

Comment:
Have you considered merging them into one? 
`| MaybeUpdateResolution (BoundedInteger -> BoundedInteger)`

``` 
MaybeUpdateResolution updateFunction ->
     debounce updateFunction model
```

then calling: `Events.onClick (MaybeUpdateResolution BoundedInteger.addOne)`